src/analysis/retail/priest/shadow/modules/tier/Tier29ShadowPriest4Set.tsx

```diff
@@ -22,13 +23,15 @@ class ShadowPriestTierSet extends Analyzer {
   protected eventHistory!: EventHistory;
   protected spellUsable!: SpellUsable;
   protected haste!: Haste;
+  has4Piece: boolean = true;
 
   uptime4Set = 0; //Buff uptime in milliseconds
   is4SetActive = false;
   timestampOfLast4SetActivation = 0; //Used to calculate uptime in case the fight ends with the buff active
 
   constructor(options: Options) {
     super(options);
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T29);
```

Comment:
It's impossible to have both the T31 4set and T29 4set at the same time - why bother with this?

Reply:
The Tier29 was always showing up in statistics even when it wasn't equipped.  I could take out the entire T29 from analysis if that would be better.

Reply:
I think you should either add a `this.active` flag to Tier29 or just remove the module outright - doesn't seem right that it will be running when almost no one will have the set equipped

Reply:
I've added `this.active` to both tier sets